src/django-app/django-project/shops/models.py

```diff
@@ -0,0 +1,31 @@
+from django.db import models
+    
+from django.utils.translation import gettext as _
+
+class Shop(models.Model):
+    shop_owner = models.ForeignKey("accounts.ShopOwner", on_delete=models.CASCADE)
+    address = models.OneToOneField("Address", on_delete=models.PROTECT)
+    name = models.CharField(_("shop name"), max_length=255, unique=True)
+    num_employees = models.IntegerField(_("number of employees"), default=0, null=True, blank=True)
+    num_bays = models.IntegerField(_("number of service bays"), default=1, null=True, blank=True)
+    availability = models.CharField(_("available hours"), max_length=255, null=True, blank=True)
+    shop_hours = models.CharField(_("shop hours"), max_length=255, null=True, blank=True)
+
+    class Meta:
+        verbose_name = "Shop"
+        verbose_name_plural = "Shops"
+
+class Address(models.Model):
+    postal_code = models.CharField(_("postal code"), max_length=6)
+    street = models.CharField(_("street"), max_length=255)
+    city = models.CharField(_("city"), max_length=255)
+    country = models.CharField(_("country"), max_length=255)
+    province = models.CharField(_("province"), max_length=2)
```

Comment:
The `province` field should allow more than 2 characters because this app may be used in countries where 2 characters may not be enough to represent a state/province.

Reply:
This is true.